Lib/contextlib.py

```diff
@@ -123,6 +128,7 @@ def __exit__(self, type, value, traceback):
                 #
                 if sys.exc_info()[1] is not value:
                     raise
+                return None
             else:
                 raise RuntimeError("generator didn't stop after throw()")
```

Comment:
Given @serhiy-storchaka's comments about readability, I'm wondering if it may make sense to drop some of the `else:` clauses in this function, such that this final `raise` ends up clearly being the final unconditionally executed line in the function. At the moment, it's far from clear that every path of execution ends in either `return` or `raise`, so it isn't actually possible to "fall off the end" and return `None`.